File: 235-lowest-common-ancestor-of-a-binary-search-tree/235-lowest-common-ancestor-of-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public TreeNode lowestCommonAncestor(TreeNode root, TreeNode p, TreeNode q) {
        int pVal = Math.min(p.val,q.val);
        int qVal = Math.max(p.val,q.val);
        return Recur(root,pVal,qVal);
    }
        
        
    public TreeNode Recur(TreeNode root, int p, int q){
        if(p<root.val && root.val<q){
            return root;
        }
        else if(p>root.val){
            return Recur(root.right,p,q);
        }
        else if(root.val>q){
            return Recur(root.left,p,q);
        }
        else if(root.val==p || root.val==q){
            return root;
        }
        return null;
    }
        
       
    
}

File: 235-lowest-common-ancestor-of-a-binary-search-tree/235-lowest-common-ancestor-of-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def lowestCommonAncestor(self, root: 'TreeNode', p: 'TreeNode', q: 'TreeNode') -> 'TreeNode':
        a = min(p.val,q.val)
        b = max(p.val,q.val)
        
        curr=root
        while(curr):
            if(not curr):
                break
            if(a<=curr.val<=b):
                return curr
            if(b<curr.val):
                curr=curr.left
            if(a>curr.val):
                curr=curr.right
                
                
                
        
        
        